src/attached_probe.cpp

```diff
@@ -223,16 +224,96 @@ std::string AttachedProbe::sanitise(const std::string &str)
   return std::regex_replace(str, std::regex("[^A-Za-z0-9_]"), "_");
 }
 
-uint64_t AttachedProbe::offset() const
+struct symbol {
+  std::string name;
+  uint64_t    start;
+  uint64_t    size;
+  uint64_t    address;
+};
+
+static int sym_name_cb(const char *symname, uint64_t start,
+                       uint64_t size, void *p)
 {
-  bcc_symbol sym;
-  int err = bcc_resolve_symname(probe_.path.c_str(), probe_.attach_point.c_str(),
-      probe_.loc, 0, nullptr, &sym);
+  struct symbol *sym = static_cast<struct symbol*>(p);
 
-  if (err)
-    throw std::runtime_error("Could not resolve symbol: " + probe_.path + ":" + probe_.attach_point);
+  if (sym->name == symname)
+  {
+    sym->start = start;
+    sym->size  = size;
+    return -1;
+  }
+
+  return 0;
+}
+
+static int sym_address_cb(const char *symname, uint64_t start,
+                          uint64_t size, void *p)
+{
+  struct symbol *sym = static_cast<struct symbol*>(p);
+
+  if (sym->address >= start && sym->address < (start + size))
+  {
+    sym->start = start;
+    sym->size  = size;
+    sym->name  = symname;
+    return -1;
+  }
+
+  return 0;
+}
```

Comment:
Do we need to write our own callbacks? Can we use bcc's symbol resolution helpers? See https://github.com/iovisor/bcc/blob/master/src/cc/bcc_syms.cc#L380-L399 

Reason is b/c `sym_address_cb` doesn't handle nested symbols and the logic can get a bit tricky: https://github.com/iovisor/bcc/blob/master/src/cc/bcc_syms.cc#L380-L399